### PR TITLE
CrayPE fix for binutils

### PR DESCRIPTION
**Pull request by lukebroskop**
Allow for multiple definitions at link time (for CrayPE)

### CODE REVIEW

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 1]
```diff
@@ -71,6 +71,15 @@ class Binutils(AutotoolsPackage, GNUMirrorPackage):
     # --disable-ld flag
     conflicts('~ld', '+gold')
 
+    def setup_environment(self, spack_env, run_env):
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/binutils/package.py]: Do you mean to use `setup_build_environment(self, env)` here?  The `setup_environment` method is a legacy function.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/binutils/package.py]: Or `setup_run_environment(self, env)`?

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Refer to previous questions/comments.

File: var/spack/repos/builtin/packages/binutils/package.py (hunk 1 above)
Comment (lukebroskop) [reply var/spack/repos/builtin/packages/binutils/package.py]: Thanks @tldahlgren , I would be adding these flags during the build step. However I did not see a reference to setup_build_environment (or setup_run_environment) in [https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#](https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#)

The current spack documentation indicates that setup_environment should be used: [https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#](https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#).

Perhaps the updated features are not documented yet or I'm looking at the wrong documents page?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/binutils/package.py]: > Thanks @tldahlgren , I would be adding these flags during the build step. However I did not see a reference to setup_build_environment (or setup_run_environment) in https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#
> 
> The current spack documentation indicates that setup_environment should be used: https://spack-tutorial.readthedocs.io/en/latest/search.html?q=setup_build_environment&check_keywords=yes&area=default#.
> 
> Perhaps the updated features are not documented yet or I'm looking at the wrong documents page?

That's good to know, thanks!  There are *no* buillt-in packages using `setup_environment` anymore.  The tutorial is out-of-date and needs to be updated.

**Comment by tldahlgren** [comment]
FWIW. I created an issue in the Spack tutorial (https://github.com/spack/spack-tutorial/issues/125) to address this problem.  Thanks for letting us know.

File: var/spack/repos/builtin/packages/binutils/package.py (hunk 1 above)
Comment (lukebroskop) [reply var/spack/repos/builtin/packages/binutils/package.py]: @tldahlgren thanks for the review, I've made your suggested change.

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 2]
```diff
@@ -71,6 +71,14 @@ class Binutils(AutotoolsPackage, GNUMirrorPackage):
     # --disable-ld flag
     conflicts('~ld', '+gold')
 
+    def setup_build_environment(self, env):
+
+        if self.spec.satisfies('%cce'):
+            setup_env.append_flags('LDFLAGS', '-Wl,-z,muldefs')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/binutils/package.py]:
```suggestion
            env.append_flags('LDFLAGS', '-Wl,-z,muldefs')
```

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 3]
```diff
@@ -71,6 +71,14 @@ class Binutils(AutotoolsPackage, GNUMirrorPackage):
...
+
+        if '+nls' in spec:
+            setup_env.append_flags('LDFLAGS', '-lintl')
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/binutils/package.py]:
```suggestion
            env.append_flags('LDFLAGS', '-lintl')
```

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Need to use the proper argument

File: var/spack/repos/builtin/packages/binutils/package.py [hunk 4]
```diff
@@ -99,7 +107,7 @@ def configure_args(self):
 
         if '+nls' in spec:
             args.append('--enable-nls')
-            args.append('LDFLAGS=-lintl')
+
```
Comment (haampie) [inline var/spack/repos/builtin/packages/binutils/package.py]:
```suggestion
```

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**